### ISSUE TITLE
Declare some quests as unimportant (show them only if all important quests are solved)

### ISSUE THREAD

**Issue by mnalis** [streetcomplete/StreetComplete#2944]
(Especially until some form of  is implemented, this might help alleviate some of noise

**Use case**
As noted in several places (most notably in "quests presets"  https://github.com/streetcomplete/StreetComplete/issues/1654 and linked issues) there are many times when user would like to reduce or increase number of quests shown. Selecting and deselecting quests to be shown manually each time is overwhelmingly time consuming and annoying to user.

Because, when all quests that the user would want to solve are enabled, in areas with less mapping activity it populates the map with way too many quests. On the other hand, when only important quests are selected, map is mostly empty in areas with more mapping activity.

Some form of managing the number of quest displayed that is achieved with much less effort than current solution (manually selecting/deselecting on quest by quest basis) is required.

**Proposed Solution**

What I propose here is to have a line delimiting "important" quests from "unimportant" ones. We currently allow the user to move the quest up and down the priority list. If we set a visible line, and declare quests above it as important, and quest below it as unimportant, then the user can move quests below or above that line. If the quests is defined as "important" in this way, then it is normally shown as before. However, if quest is defined as "unimportant" (ie. is moved below the demarcation line), then such quests should be shown only if there are no "important" quests being shown on the map.

Unimportant quests could be colored with transparency for more clarity, for example this mockup: 
![importance_line2](https://user-images.githubusercontent.com/156656/120687734-947e8e00-c4a2-11eb-8451-d7e2f763d51b.jpg)

This would allow user interface to remain mostly as it is, thus providing users with added value (no need to go to enable/disable quests all the time in order to have reasonable number of quests that user wants displayed) at no added (user complexity) cost. 

Notes/Alternatives:
- of course, there is still a programming cost (and I'm not familiar with codebase to guess how complex it is, or even if libraries used allow for such customizations)
- Alternative: Fully customizable quest presets with quick-change as mentioned in https://github.com/streetcomplete/StreetComplete/issues/1654#issuecomment-745653725 would IMHO be the best way. However, for fully customizable user presets there would be more initial complexity for users wanting to use it (they would need to setup several full lists of quests, which is time consuming) - while not as powerful, this proposal requires less setup time (only one list to setup) and is arguably simpler. Also, both could be implemented as they are somewhat complementary.
- Alternative: have users define list of quests they want to solve in exactly the same way as now, but also provide quick toggle to `show all quests` / `show only selected quests` as mentioned in https://github.com/streetcomplete/StreetComplete/issues/1301. It would allow users to optimize for less-mapped areas, and when they get to more-mapped areas where the map shows no quests, they would quickly click `show all quests` toggle to show all the remaining quests. Would give user more manual control than this idea (and would be trivial to implement!), but would require user to click the toggle while they move in different areas (while this suggestion would work automatically once setup). This is also complementary, ie. can be used as only feature implemented, or in combination with other alternatives.

**Comment by mnalis** [comment]
My order of preference (most wanted at top) would be:
- fully customizable user presets (https://github.com/streetcomplete/StreetComplete/issues/1654#issuecomment-745653725)
- simple `show all quests` / `show only selected quests` toggle (https://github.com/streetcomplete/StreetComplete/issues/1301)
- demarcation line for unimportant quests (this issue -  https://github.com/streetcomplete/StreetComplete/issues/2944) 

(Incidently, this is also the order of giving the user most control over customization.)

**Comment by matkoniecz** [comment]
> populates the map with way too many quests

What is exactly the problem here? When you solve quests ones more important are already shown in preference to less important ones (except cases where even more important is blocking there display).

Is it "with zoomed out map many dots are shown, on zooming in or visiting location it turns to be asking about power pole material/bench backrest"?

**Comment by matkoniecz** [comment]
Well explained what you propose (one missing part is asked in a previous comment).

One of main problems (in addition that as you mentioned it will not code itself) with this idea is a potential confusion. How to explain what is going on in quest selection interface? It is already quite confusing and complex.

Also, as map is dragged all quests could suddenly vanish without a clear reason.

For maximum confusion scenario: more important quest is in range but just below finger of user on a screen. It is displayed so all other quests are hidden.

**Comment by smichel17** [comment]
- This sounds like a simpler version of what I originally proposed in https://github.com/streetcomplete/StreetComplete/issues/1654#issuecomment-741970370 and discussed in more detail starting at https://github.com/streetcomplete/StreetComplete/discussions/2457#discussioncomment-267405. Since it is based only on the density of quests, not the user's current position, I think it not be blocked on the Tangram issue like my suggestion is.

- I don't think the line is needed to implement something like this. Instead, only display quests if there are fewer than N quests already displayed nearby, where N is chosen with a sliding scale based on how far down the priority list that quest is ⇒ quests at the top are shown with arbitrary density; quests at the bottom only appear when the density is low enough. Probably the only new UI which is needed is a switch to determine whether this mode should be engaged or whether all quests should be displayed.

**Comment by mnalis** [comment]
@matkoniecz the main problem I see is that (default) set of quest presents **overwhelming** number of quests (in slightly mapped areas), majority of which are of low importance.  This is highly off-putting to the user (especially as they see most of them are of low importance), and thus demotivating. 

The user can work around it by deselecting quests of low importance after installing the app, but that:

- presents an additional effort for each user (and needs them to be aware of the possibility), which we want to avoid (as we value each users time)
- creates a new problem when user manages to solve important quests in an area, because "low importance" is not the same as "completely useless", and users should be presented with low importance quests (instead of empty map) if there is nothing better for them to do.
- to work around that second issue (if they put the effort in working around the first issue as detailed), users can currently only manually enable/disable quest by quest **each and every time** they change location (depending on how well is he new area mapped), which is highly time consuming and demotivating too, which we also (presumably) want to avoid.

But you do raise **very important** point of this suggestion potentially being confusing the users, which I completely agree is big problem (especially after being befuddled myself for a long time because of UI issues like https://github.com/streetcomplete/StreetComplete/issues/2817).

So, I'll leave it to @westnordost to close this issue then, unless he sees some merit in this approach, or someone else can think of a way how to make it clear to user what is happening and eliminate possible confusion. (One of the [remaining options](https://github.com/streetcomplete/StreetComplete/issues/2944#issuecomment-854067839) to handle overwhelming number of quests is probably a better solution then)

**Comment by Lee-Carre** [comment]
This may warrant its own issue, but seems relevant.

I've tweaked the quest list in a few places.

One problem is when the answer to the lane-count quest is ambiguous (between no marked lanes versus only a single lane), yet answering the likes of sidewalk or cycleway is easy. When the lanes quest is higher priority, it blocks the quests which are easier to definitively answer.

Another is being asked if a way is lit, while it's obviously still daytime, and not being asked quests which can be answered at any time of day. Being able to see street-lamps doesn't necessarily mean that the way is sufficiently lit; some roads have sections which are dark (from widely spaced and/or dim lighting).

So, I had thought that asking if a way is lit should be conditional (only ask at night, and with a higher priority than during daylight, while prioritising other quests during daytime surveying).

My attitude is one of; if this is supposed to be source=survey, then I can't possibly answer the question until it's dark and the lamps are on. It's not even simply sunset-sunrise; they seem to trigger based on ambient light intensity.

It doesn't seem infeasible to make a reasonable guess of when the threshold occurs; I can think of several ways. Failing that, make it user-customisable (e.g. only ask me about lighting after 22:00 and before 06:00). Else, make it depend on the theme-change configuration.

**Comment by mnalis** [comment]
As for the `lit` quest, see https://github.com/streetcomplete/StreetComplete/issues/1285

**Comment by westnordost** [comment]
Alright, I will read again through all that has been said and proposed before, consider it and implement something. Though, be warned that the "something" will most probably not cover all the use cases mentioned in this and previous discussions.

The most important use case, that I want to cover, is to not overwhelm users with spammy unimportant quests that feel like work (building type, building levels etc) because this is bad for user retention and is opposed to this app's mission to be something for the "casual" mapper.

So the final solution I might select might be as easy as disabling annoying quests by default, or showing a simple dialog in the beginning that offers to do this and point the user towards that he can change that in the settings. We'll see.

Also, most likely, I will not implement different profiles (f.e. in car, on bike etc) and also no custom profiles because this is a feature rather for power users and power users will have less of a problem to spend a few minutes checking or unchecking every box in that settings menu.

**Comment by matkoniecz** [comment]
> showing a simple dialog in the beginning that offers to do this and point the user towards that he can change that in the settings. We'll see.

Maybe show that dialog after 20/50/100 edits? So that start will not be explosion of dialogs, menus and wizards?

**Comment by matkoniecz** [comment]
> spammy unimportant quests that feel like work (building type, building levels etc)

At least for me building levels and roof shape feels quite spammy and pointless, but I am fine with building type. More interesting, actually useful info (also in StreetComplete itself - address quest).

I would keep it enabled by default.

**Comment by mnalis** [comment]
> Also, most likely, I will not implement different profiles (f.e. in car, on bike etc) and also no custom profiles because this is a feature rather for power users and power users will have less of a problem to spend a few minutes checking or unchecking every box in that settings menu.

I understand (and even **support**, as much as I would _really want to see custom profiles_) such decision to cater to **novice users** instead of power users - that is main selling point of SC, truly, and should have **priority always**. 

That being said, I do not really agree that such timewasting/annoying behaviour in SC is more acceptable (_"less of a problem"_) to power users than to novice ones. If anything, heavy poweruser would (by definition) engage in such selecting/deselecting **much more often** than casual novice user (who would not be going around clicking on preferences all the time, if at all) - and by sheer amount of such extra clicking work thus be at last order or two of magnitude more annoyed (because for them, that _"few minutes of extra clicking"_ in preferences will be happening every few dozen minutes, and not once in a blue moon - as it would for casual novice).

**Comment by smichel17** [comment]
> Alright, I will read again through all that has been said and proposed before, consider it and implement something.

@westnordost Filtering is an area I'd be willing to help implement, especially something along the lines of what we discussed in https://github.com/streetcomplete/StreetComplete/discussions/2457#discussioncomment-267405 or the initial implementation of https://github.com/streetcomplete/StreetComplete/issues/2565\*. However, I have a couple of other projects I want to do first, so I will not get to it until *at least* August 2021; I'd support something simple and temporary like disabling spammier quests (building height, roof shape) by default.

\*I apologize that I haven't gotten around to summarizing it yet, so the discussion is still quite long, but the first post describes the idea of the initial implementation well enough.

**Comment by westnordost** [comment]
No thanks, I will do it myself.

**Comment by westnordost** [comment]
But thank you for the links again, I'll read through them again!

**Comment by andrewharvey** [comment]
OSM doesn't have important and less important tags, it just has tags, what's important or not is subjective and depends on the contributor and data consumer. In my opinion StreetComplete should accept any valid quest based on mappers being able to answer the quest and the tag being well supported (as a project it should, but of course no individual as a volunteer should have to review such a PR they aren't interested in).

If we wan't to try to ease in new comers by all means group quests into beginner and old timer and only show the beginner ones by default, but still let users choose to turn those on and tweak them.

**Comment by mnalis** [comment]
@andrewharvey I agree with premise of "importance is in the eye of the beholder", and that the user should be able to [manually enable/disable](https://github.com/streetcomplete/StreetComplete/issues/2944#issuecomment-854067839) things that are important to **them**. Issue here IMHO is that with rising number of quests (already pretty high):
- new users are overwhelmed, and many of the quests will look useless to them, so that will lead to uninstalling the app; while if they had feeling it is asking useful information (and number of presented quests was more managable), they would continue to use it and contribute
- even for power users, current system of managing quests is very time consuming, even if they know what each tag does by heart (and do not have to consult the wiki or look at the code). Just going through all quests one by one and reading descriptions is lots of work. Now imagine if we had 10 thousand quests (still just a small part of used keys in OSM). Surely not of them should be present in mainstream SC by default? Practically no one would be willing to go through such list to select interesting things. So things should be grouped somehow (imagine if stuff in supermarket was put on shelves randomly, and not grouped!)
  So this issue suggests to create two groups, based on their usefulness  _as perceived by **devels**_ (_hopefully_ matching that of the **majority** of users - but of course there would be users with different view of what is important to them, and thus forced to do more clicking to select stuff interesting to them). Other (more customizable) options would be to create more than two groups of quests based on other criterias (as suggested in [customs presets](https://github.com/streetcomplete/StreetComplete/issues/1654#issuecomment-745653725) for example), but the devels do not (currently) seem too keen on that idea.

**Comment by westnordost** [comment]
In preparation for deciding how this can be solved, I created this board.

https://miro.com/welcomeonboard/UnBNenVoV1dYT3FpeThOdnJaaU10ZjE2UWxYVThlQ0YwZTN2akVYZThVYXRUakVockl0S0l1Rm1Ra0Vyb3RLWXwzMDc0NDU3MzQ5NTk3NDI5MzAy

We already have the order "by importance". I sorted the quests in this grid roughly by occurances by answer difficulty and time

**Comment by matkoniecz** [comment]
> In preparation for deciding how this can be solved, I created this board.

I am not entirely sure: are you welcoming people to edit it? All of them or just some? Or leave feedback here? Or just look at the progress?

**Comment by westnordost** [comment]
You are welcome to edit it and leave comments. The "how many occurances" however is something I took roughly from the quest statistics

**Comment by westnordost** [comment]
close as will not fix / solved kind of differently in #3210
